src/resources/Clusters/Agents/tests/ClusterAgent.spec.ts

```diff
@@ -18,19 +16,19 @@ describe('clusterAgent', () => {
     });
 
     describe('list', () => {
-        it('should make a GET call to the ClusterAgent base url', () => {
-            clusterAgent.list(clusterId);
+        it('should make a GET call to the ClusterAgent base url', async () => {
+            await clusterAgent.list(clusterId);
 
             expect(api.get).toHaveBeenCalledTimes(1);
             expect(api.get).toHaveBeenCalledWith(ClusterAgent.getBaseUrl(clusterId));
         });
     });
 
     describe('upgrade', () => {
-        it('should make a PUT call to the upgrade cluster agent url', () => {
+        it('should make a PUT call to the upgrade cluster agent url', async () => {
             const agentId = '👐';
             const data: ClusterNodeUpgradeDataModel = {componentName: 'butterfly', version: '1.2.3'};
-            clusterAgent.upgrade(clusterId, agentId, data);
+            await clusterAgent.upgrade(clusterId, agentId, data);
```

Comment:
If this is just linting and typing... I wonder how come we get this kind of occurrences where we add asynchronicity handling where we did not need it before. This sounds like more than linting and typing to me. I think I would feel more confident if this big PR full of changes wouldn't obviously modify how the code works. Do we know what is causing this, is it really just ESLINT deciding what we should async/await ?

Reply:
This rule is quite nice actually: https://typescript-eslint.io/rules/no-floating-promises/. Basically all our tests were leaking unresolved promises.